make_beam/make_beam.c

```diff
@@ -358,274 +344,155 @@ int main(int argc, char **argv)
 
     // Set up parrel streams
     cudaStream_t streams[npointing];
-
     for ( p = 0; p < npointing; p++ )
         cudaStreamCreate(&(streams[p])) ;
-    
-    // TODO work out why the below won't work. It should save a bit of time 
-    // instead of doing the same thing every second
-    //populate_weights_johnes( &gf, complex_weights_array, invJi,
-    //                         npointing, nstation, nchan, npol );
-        
 
     fprintf( stderr, "[%f]  **BEGINNING BEAMFORMING**\n", NOW-begintime);
-    
-    // Set up sections checks that allow the asynchronous sections know when 
-    // other sections have completed
-    int file_no;
-    int *read_check;
-    int *calc_check;
-    int **write_check;
-    read_check = (int*)malloc(nfiles*sizeof(int));
-    calc_check = (int*)malloc(nfiles*sizeof(int));
-    write_check = (int**)malloc(nfiles*sizeof(int *));
-    for ( file_no = 0; file_no < nfiles; file_no++ )
-    {
-        read_check[file_no]  = 0;//False
-        calc_check[file_no]  = 0;//False
-        write_check[file_no] = (int*)malloc(npointing*sizeof(int));
-        for ( p = 0; p < npointing; p++ ) write_check[file_no][p] = 0;//False
-    } 
-    
+
     // Set up timing for each section
-    long read_total_time, calc_total_time, write_total_time;
+    long read_time[nfiles], delay_time[nfiles], calc_time[nfiles], write_time[nfiles][npointing];
+    clock_t start;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'start' can be reduced.](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=5062919)